docs/guides/deployment-patterns.md

```diff
@@ -0,0 +1,170 @@
+---
+description: Answers to frequently asked questions about Prefect.
+tags:
+    - FAQ
+    - frequently asked questions
+    - questions
+    - license
+    - databases
+---
+
+
+
+
+
+### Patterns in Prefect
+There are four common dataflow design patterns in Prefect. Each pattern offers different degrees and types of separation from related flows.
+
+ - Conceptual separation is when a flow can be thought of as separate from another flow, even if it’s part of the same process.
+
+ - Execution separation is when a flow can be executed separately from another flow.
+
+ - Awareness separation is when a flow doesn’t have any direct reference to another flow, even if it’s related.
```

Comment:
are these concepts taken from someplace or did we make up this language? I dont think these add a lot in terms of user understanding of deployment patterns, personally I'd want to remove these bullets + their references below

Reply:
Makes sense, I pulled them from Bill's blog on [deployment patterns](https://www.prefect.io/guide/blog/dataflow-design-patterns/). A lot of the content is directly pulled from there, just thought having this in the docs would be a better home (and would get some more eyes on it too).

Reply:
I can remove the bulleted points if its unnecessary

Reply:
ahh yeah I see that. I'm open to keeping them if we think that makes more sense, would defer to @billpalombi. I just find the bullets a little confusing